workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/androidx/StateRegistryAggregator.kt

```diff
@@ -0,0 +1,300 @@
+   *
+   *  - if [view] is dropped but may be restored before [detachFromParentRegistry] is called,
+   *    as when pushing and popping a back stack, call [saveChildRegistryOwner] to capture
+   *    its state tear down its [ViewTreeSavedStateRegistryOwner]. The saved state
+   *    will be restored to the next [view] passed to [installChildSavedStateRegistryOwnerOn]
+   *    with the same [key]
+   *
+   *  - if [view] is dropped and will not be restored, as when a window is closed or
+   *    back stack history is modified, call [prune] _with the keys of the views that
+   *    remain active_.
+   *
+   * @param key identifier for the new [ViewTreeSavedStateRegistryOwner], unique across this
+   * [StateRegistryAggregator]. Typically this is derived from the
+   * [compatibility key][com.squareup.workflow1.ui.Compatible.keyFor] of the [view]'s
+   * rendering.
+   */
+  public fun installChildSavedStateRegistryControllerOn(
+    view: View,
+    key: String
+  ) {
+    val lifecycleOwner = checkNotNull(WorkflowLifecycleOwner.get(view)) {
+      "Expected container view to have set a WorkflowLifecycleOwner on its immediate " +
+        "child $view."
+    }
+    val registryOwner = KeyedStateRegistryOwner(key, lifecycleOwner)
+    children.put(key, registryOwner)?.let {
+      throw IllegalArgumentException("$key is already in use, it cannot be used register $view")
+    }
+    ViewTreeSavedStateRegistryOwner.get(view)?.let {
+      "$view already has ViewTreeSavedStateRegistryOwner: $it"
+    }
+    ViewTreeSavedStateRegistryOwner.set(view, registryOwner)
+    restoreChildSavedStateRegistryOwnerIfReady(registryOwner)
+  }
+
+  /**
+   * Call this when the [View] previously [registered][installChildSavedStateRegistryControllerOn]
+   * with [key] is being dropped, but may be replaced with a new instance before the container
+   * is destroyed -- think of pushing and popping in a back stack.
+   *
+   * The saved state will be restored if a new [View] is
+   * [registered][installChildSavedStateRegistryControllerOn] with the same [key].
+   */
+  public fun saveChildSavedStateRegistryOwner(key: String) {
+    children.remove(key)?.let { saveChildSavedStateRegistryOwner(it) }
+      ?: throw IllegalArgumentException("No such child: $key")
+  }
+
+  private fun saveChildSavedStateRegistryOwner(child: KeyedStateRegistryOwner) {
+    doIfRestored { states ->
+      val state = Bundle()
+      child.controller.performSave(state)
+      states += child.key to state
+    }
+  }
+
+  /**
+   * If this object has been restored from its parent registry, restores the child
+   * [KeyedStateRegistryOwner.controller] that was previously [saved][saveChildSavedStateRegistryOwner]
+   * with the same [KeyedStateRegistryOwner.key].
+   *
+   * This method can only be called once per key per instance of this class.
+   * After a key has been restored, its data will be removed from this object.
```

Comment:
Is this part important? Why do we remove it? IS that what AndroidX is doing as well?

Reply:
This is a lie, I deleted it. We actually do call this method more than once.